backend/events/views.py

```diff
@@ -1,3 +1,15 @@
-from django.shortcuts import render
+from rest_framework import generics
+
+from .models import Event
+from .serializers import EventRegistrationSerializer, EventSerializer
 
 # Create your views here.
+
+
```

Comment:
@HarshRaj2717 the purpose of using CBVs is to clump together the views associated to single entity

we should clump together these tow as well
and also overriding `get_serializer_class` for the flexibility 
so we can do something like this.....

`mixins` behind the scene will from your `get` `post`  definitions calls `list` `create` `update` `delete` calls and utilises other functions which can be overriden and this is single class to be dealt with
more robust in nature and requires less changes in future if change serializers.

check out https://www.django-rest-framework.org/tutorial/3-class-based-views/#using-mixins
  
```python

class EventAPIView(generics.GenericAPIView, 
                   ListModelMixin, 
                   CreateModelMixin):
    queryset = Event.objects.all()
    serializer_class = EventSerializer  # Serializer for listing events

    def get_serializer_class(self):
        if self.request.method == 'POST':
            return EventRegistrationSerializer  #  Or perhaps avoid the whole serialize_class assingment since this will handle it 
        return EventSerializer

    def get(self, request, *args, **kwargs):
        return self.list(request, *args, **kwargs)

    def post(self, request, *args, **kwargs):
        return self.create(request, *args, **kwargs)
```

Reply:
Updated it 👍